File: project/cs/v3cs/v1cs/PrintScanner.cs
/***********************************************************************
 * Module:  PrintScanner.cs
 * Author:  Nabra
 * Purpose: Definition of the Class PrintScanner
 ***********************************************************************/

using System;

public class PrintScanner
{
   public void AssesOfPrinter()
   {
      // TODO: implement
   }
   
   public void AssesOfScaner()
   {
      // TODO: implement
   }
   
   public void PutPaper()
   {
      // TODO: implement
   }
   
   public void PutPaint()
   {
      // TODO: implement
   }
    /// <summary>
    /// Конструктор за замовченням
    /// </summary>
   public PrintScanner()//конструктор за замовченням
   {
       stayOfPrinter = false;
       stayOfScanner = false;
       numbPaper = 0;
       numbPaint = 0;
       maxPaint = 0;
       maxPaper = 0;
       Console.WriteLine("PrintScanner created standartly");
   }

    /// <summary>
    /// Конструктор копії
    /// </summary>
    /// <param name="prin">Об*єкт, що копіюється</param>
   public PrintScanner(PrintScanner prin)
{
	//атрибути отриманого класу визначаються атрибтами вхідного класу
	stayOfPrinter = prin.stayOfPrinter;
	stayOfScanner = prin.stayOfScanner;
	numbPaper = prin.numbPaper;
	numbPaint = prin.numbPaint;
	maxPaint = prin.maxPaint;
	maxPaper = prin.maxPaper;
    Console.WriteLine("PrintScanner created by copy");
}
   private bool stayOfPrinter { get; set; }
   private int numbPaper
   {
       get
       {
           return numbPaper;
       }
       set
       {
           if (value < 0)
               numbPaper = 0;
           else
               numbPaper = value;
       }
   }
   private int numbPaint
   {
       get
       {
           return numbPaint;
       }
       set
       {
           if (value < 0)
               numbPaint = 0;
           else
               numbPaint = value;
       }
   }

   private int maxPaper
   {
       get
       {
           return maxPaper;
       }
       set
       {
           if (value < 0)
               maxPaper = 0;
           else
               maxPaper = value;
       }
   }
   private int maxPaint
   {
       get
       {
           return maxPaint;
       }
       set
       {
           if (value < 0)
               maxPaint = 0;
           else
               maxPaint = value;
       }
   }
   private bool stayOfScanner { get; set; }

}

File: project/c++/V4/Version1/Version1/PrintScanner.cpp
/***********************************************************************
 * Module:  PrintScanner.cpp
 * Author:  Nabra
 * Modified: 17 октября 2016 г. 17:56:57
 * Purpose: Implementation of the class PrintScanner
 ***********************************************************************/

#include "PrintScanner.h"
#include <iostream>
using namespace std;

//=========== Конструктор за замовченням ===========
PrintScanner::PrintScanner()
{
	stayOfPrinter = false;
	stayOfScanner = false;
	numbPaper = 0;
	numbPaint = 0;
	maxPaint = 0;
	maxPaper = 0;
	cout << "PrintScanner created standartly" << endl;
}

//=========== Конструктор копії ===========
PrintScanner::PrintScanner(const PrintScanner&prin)
{
	//атрибути отриманого класу визначаються атрибтами вхідного класу
	stayOfPrinter = prin.stayOfPrinter;
	stayOfScanner = prin.stayOfScanner;
	numbPaper = prin.numbPaper;
	numbPaint = prin.numbPaint;
	maxPaint = prin.maxPaint;
	maxPaper = prin.maxPaper;
	cout << "PrintScanner created by copy" << endl;
}

PrintScanner::~PrintScanner()
{
	cout << "PrintScanner distroed" << endl;
}

//визначити справність принтера
bool PrintScanner::getPrinter()
{
	return stayOfPrinter;
}

//визначити справність сканера
bool PrintScanner::getScanner()
{
	return stayOfScanner;
}

//повернути кількість паперу
int PrintScanner::getPaper()
{
	return numbPaper;
}

//повернути макс. кількість паперу
int PrintScanner::getMPaper()
{
	return maxPaper;
}

//повернути макс. кількість фарби
int PrintScanner::getMPaint()
{
	return maxPaint;
}

////////////////////////////////////////////////////////////////////////
// Name:       PrintScanner::AssesOfPrinter()
// Purpose:    Implementation of PrintScanner::AssesOfPrinter()
// Return:     void
////////////////////////////////////////////////////////////////////////

void PrintScanner::AssesOfPrinter(bool asses)
{
	stayOfPrinter = asses;
}

////////////////////////////////////////////////////////////////////////
// Name:       PrintScanner::AssesOfScaner()
// Purpose:    Implementation of PrintScanner::AssesOfScaner()
// Return:     void
////////////////////////////////////////////////////////////////////////

void PrintScanner::AssesOfScaner(bool asses)
{
	stayOfScanner = asses;
}

////////////////////////////////////////////////////////////////////////
// Name:       PrintScanner::PutPaper()
// Purpose:    Implementation of PrintScanner::PutPaper()
// Return:     void
////////////////////////////////////////////////////////////////////////

//додати фарбу
void PrintScanner::PutPaper(int numb/*повернути кількість доданої фарби*/)
{
	if (numb <= maxPaper)
		numbPaper += numb;
	else
		numbPaper = maxPaper;
}

////////////////////////////////////////////////////////////////////////
// Name:       PrintScanner::PutPaint()
// Purpose:    Implementation of PrintScanner::PutPaint()
// Return:     void
////////////////////////////////////////////////////////////////////////

//додати папір
void PrintScanner::PutPaint(int numb/*кількість доданого паперу*/)
{
	if (numb <= maxPaint)
		numbPaint += numb;
	else
		numbPaint = maxPaint;
}

//повернути макс. кількість паперу
void PrintScanner::setMPaper(int numb)
{
	maxPaper = numb;
}

//повернути макс. кількість фарби
void PrintScanner::setMPaint(int numb)
{
	maxPaint = numb;
}
